src/orbit.rs

```diff
@@ -239,6 +254,70 @@ where
     }
 }
 
+pub struct Orbits<O>
+where
+    O: Orbit,
+{
+    pub stores: RwLock<BTreeMap<Cid, O>>,
+    pub base_path: PathBuf,
+}
+
+#[rocket::async_trait]
+pub trait OrbitCollection<O: Orbit> {
+    async fn orbits(&self) -> RwLockReadGuard<BTreeMap<Cid, O>>;
+    async fn add(&self, orbit: O) -> ();
+}
+
+impl<O: Orbit> Orbits<O> {
+    fn new<P: AsRef<Path>>(path: P) -> Self {
+        Self {
+            stores: RwLock::new(BTreeMap::new()),
+            base_path: path.as_ref().to_path_buf(),
+        }
+    }
+}
+
+#[rocket::async_trait]
+impl<O: Orbit> OrbitCollection<O> for Orbits<O> {
+    async fn orbits(&self) -> RwLockReadGuard<BTreeMap<Cid, O>> {
+        self.stores.read().await
+    }
+
+    // fn orbit(&self, id: &Cid) -> Option<&O> {
```

Comment:
should these be called `get` for consistency with `add` (and possibly `remove` in the future)?

Reply:
they probably should (as well, `orbits.orbit` doesnt read well), but until a better solution for the store types/lifetimes is possible I think we can wait